### PR TITLE
Refactor sanitizer skip tests into test.support

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -382,6 +383,34 @@ def skip_if_buildbot(reason=None):
...
+        reason = 'not working with sanitizers active'
+
+    _cflags = sysconfig.get_config_var('CFLAGS') or ''
+    _config_args = sysconfig.get_config_var('CONFIG_ARGS') or ''
+    memory_sanitizer = (
+        '-fsanitize=memory' in _cflags or
+        '--with-memory-sanitizer' in _config_args
+    )
+    address_sanitizer = (
+        '-fsanitize=address' in _cflags
```
Comment (erlend-aasland) [inline Lib/test/support/__init__.py]: Also check `'--with-memory-sanitizer'` in `_config_args`

**Review by erlend-aasland — APPROVED** [review]
Nice addition. LGTM, with a single remark.

**Comment by erlend-aasland** [comment]
I'd add a NEWS entry and possibly document it in the test support section.

File: Lib/test/support/__init__.py [hunk 2]
```diff
@@ -55,6 +55,7 @@
     "run_with_tz", "PGO", "missing_compiler_executable",
     "ALWAYS_EQ", "NEVER_EQ", "LARGEST", "SMALLEST",
     "LOOPBACK_TIMEOUT", "INTERNET_TIMEOUT", "SHORT_TIMEOUT", "LONG_TIMEOUT",
+    "skip_if_sanitizer",
```
Comment (erlend-aasland) [inline Lib/test/support/__init__.py]: Nit: this probably belongs in the `unittest` section between line 34 and 44.

**Comment by pablogsal** [comment]
> I'd add a NEWS entry and possibly document it in the test support section.

The news section is for user visible changes, this is an internal change.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**